### PR TITLE
Improve GithubRepositoryProvider.getUserRepos – EXP-407

### PR DESCRIPTION
**Pull request by AlexTugarev**
The graphql query is extend to mimic what you would see in Top Repositories box visiting github.com.

## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at ca89a5a</samp>

Rewrite `getUserRepos` method in `github-repository-provider.ts` to use a more complex GraphQL query that fetchs and categorizes the user's repositories. This improves performance, accuracy, and supports a new UI feature.

</details>

## Related Issue(s)

Fixes EXP-407

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-gh-getuserrepos</li>
	<li><b>🔗 URL</b> - <a href="https://at-gh-getuserrepos.preview.gitpod-dev.com/workspaces" target="_blank">at-gh-getuserrepos.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - at-gh-getuserrepos-gha.14977</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by AlexTugarev** [comment]
The query in plain code:

```
fragment Repos on RepositoryConnection {
	nodes {
	  url
	}
  }
  
  query topRepositories {
	viewer {
	  contributedTo: repositoriesContributedTo(
		first: 100
		orderBy: {field: PUSHED_AT, direction: DESC}
		includeUserRepositories: true
		contributionTypes: [COMMIT]
	  ) {
		...Repos
	  }
	  original: repositories(
		first: 100
		ownerAffiliations: OWNER
		privacy: PUBLIC
		isFork: false
		isLocked: false
		orderBy: {field: UPDATED_AT, direction: DESC}
	  ) {
		...Repos
	  }
	  forked: repositories(
		first: 100
		ownerAffiliations: OWNER
		privacy: PUBLIC
		isFork: true
		isLocked: false
		orderBy: {field: UPDATED_AT, direction: DESC}
	  ) {
		...Repos
	  }
	}
  }
```

File: components/server/src/github/github-repository-provider.ts [hunk 1]
```diff
@@ -172,28 +172,57 @@ export class GithubRepositoryProvider implements RepositoryProvider {
...
+                    orderBy: {field: PUSHED_AT, direction: DESC}
+                    includeUserRepositories: true
+                    contributionTypes: [COMMIT]
+                  ) {
+                    ...Repos
+                  }
+                  original: repositories(
+                    first: 100
+                    ownerAffiliations: OWNER
+                    privacy: PUBLIC
```
Comment (filiptronicek) [inline components/server/src/github/github-repository-provider.ts:197]: Is this by design that we only take public repos? Any benefits in doing that?

Comment (AlexTugarev) [reply components/server/src/github/github-repository-provider.ts:197]: Nice catch. Going to drop that line.

**Comment by AlexTugarev** [comment]
Strange, it looks like it's not even called 🙈 
cc. @akosyakov

**Pull request merged**